tests/test_dagger.py

```diff
@@ -113,35 +109,33 @@ def test_trainer_makes_progress(tmpdir, session):
 
 
 def test_trainer_save_reload(tmpdir, session):
-    with tf.variable_scope("orig_trainer"):
-        trainer = make_trainer(tmpdir)
+    trainer = make_trainer(tmpdir)
     trainer.round_num = 3
     trainer.save_trainer()
-    with tf.variable_scope("reload_trainer"):
-        new_trainer = trainer.reconstruct_trainer(tmpdir)
+    new_trainer = dagger.reconstruct_trainer(tmpdir)
     assert new_trainer.round_num == trainer.round_num
 
     # old trainer and reloaded trainer should have same variable values
-    old_vars = trainer._sess.run(trainer._vars)
-    new_vars = trainer._sess.run(new_trainer._vars)
-    assert len(old_vars) == len(new_vars)
-    for v1, v2 in zip(old_vars, new_vars):
-        assert np.allclose(v1, v2)
+    old_vars = trainer.bc_trainer.policy.state_dict()
+    new_vars = new_trainer.bc_trainer.policy.state_dict()
+    assert len(new_vars) == len(old_vars)
+    for var, values in new_vars.items():
+        assert values.equal(old_vars[var])
 
     # also those values should be different from a newly created trainer
-    with tf.variable_scope("third_trainer"):
-        third_trainer = make_trainer(tmpdir)
-    third_vars = trainer._sess.run(third_trainer._vars)
-    all_same = True
-    for v1, v3 in zip(old_vars, third_vars):
-        all_same = all_same and np.allclose(v1, v3)
-    assert not all_same
+    third_trainer = make_trainer(tmpdir)
+    third_vars = third_trainer.bc_trainer.policy.state_dict()
+    assert len(third_vars) == len(old_vars)
+    difference = False
```

Comment:
I think this can be simplified to:

```python
assert any((values.equal(old_vars[var]) for var, values in third_vars.items())
```

Reply:
Nice! I just simplified it to `assert not any(values.equal(old_vars[var]) for var, values in third_vars.items())`